File: javascript/gigasecond/gigasecond.js
module.exports = function (expulsionDay) {
  'use strict';

  var GIGASECOND = Math.pow(10, 9);

  this.date = function () {
    var happyDay = expulsionDay;
    happyDay.setSeconds(happyDay.getSeconds() + GIGASECOND);
    happyDay.setHours(0, 0, 0, 0);

    return happyDay;
  };
};

File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Calculates the moment when someone has lived for a gigasecond (10^9 seconds).
 */
public class Gigasecond {
  static LocalDateTime date;
  static long GIGASECOND = (long)Math.pow(10, 9);

  Gigasecond(LocalDate birthdate) {
    this.date = birthdate.atTime(0, 0).plusSeconds(GIGASECOND);
  }

  Gigasecond(LocalDateTime birthdate) {
    this.date = birthdate.plusSeconds(GIGASECOND);
  }

  public final LocalDateTime getDate() {
    return this.date;
  }
}
